src/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -36,7 +46,16 @@ public MSBuildForwardingApp(IEnumerable<string> argsToForward)
 
         public int Execute()
         {
-            return _forwardingApp.Execute();
+            try
+            {
+                Environment.SetEnvironmentVariable(TelemetrySessionIdEnvironmentVariableName, Telemetry.CurrentSessionId);
```

Comment:
Who are the intended consumers of this env var? If it is just the MSBuildLogger constructor, would it be cleaner to set it as a static on that type instead? Remember the MSBuild tests and environment variables :)